Shared/Extensions/NSURLExtensions.swift

```diff
@@ -318,13 +318,27 @@ extension URL {
 // Extensions to deal with ReaderMode URLs
 
 extension URL {
+    // Is this an internal ReaderMode url?
     public var isReaderModeURL: Bool {
         let scheme = self.scheme, host = self.host, path = self.path
         return scheme == "http" && host == "localhost" && path == "/reader-mode/page"
     }
 
+    // Is this a synced ReaderMode url? about:reader?url=http://...
+    public var isSyncedReaderModeURL: Bool {
+        if self.scheme != "about" {
+            return false
+        }
+        let urlParts = self.absoluteString.components(separatedBy: "?url=")
```

Comment:
Instead of splitting the string like this in NSURLExtensions there should be a getQuery() method. You can use it to split the query into a dictionary and then simply do `let  readerURL = self.getQuery()['url']`